src/RileyMaze/RMZUnit.js

```diff
@@ -0,0 +1,219 @@
+    }
+
+    _onActiveChanged(childName, button) {
+        this._model.set_property(childName, directions[button.get_active()]);
+    }
+
+    bindWindow(win) {
+        win.lockscreen.key = 'item.key.rileymaze.1';
+        win.lockscreen.lock = 'lock.rileymaze.1';
+        void this;
+    }
+
+    _compile() {
+        const code = this._variablesCodeview.text;
+
+        if (code === '')
+            return;
+
+        const scope = {
+            robotADirection: null,
+            robotBDirection: null,
+        };
+        try {
+            // eslint-disable-next-line no-new-func
+            const func = new Function('scope', `with(scope){\n${code}\n;}`);
+            func(scope);
+        } catch (e) {
+            if (!(e instanceof SyntaxError || e instanceof ReferenceError))
+                throw e;
+            this._variablesCodeview.setCompileResultsFromException(e);
+            return;
+        }
+
+        if (Object.getOwnPropertyNames(scope).every(prop => prop === null))
+            return;
+
+        const errors = this._searchForErrors(scope);
+        if (errors.length > 0) {
+            this._variablesCodeview.setCompileResults(errors);
+            return;
+        }
+
+        this._variablesCodeview.setCompileResults([]);
+
+        // Block the normal notify handler that updates the code view, since we
+        // are propagating updates from the codeview to the GUI. Instead,
+        // connect a temporary handler that lets us know if anything actually
+        // did change.
+        GObject.signal_handler_block(this._model, this._notifyHandler);
+
+        let guiUpdated = false;
+        const tempHandler = this._model.connect('notify', () => {
+            guiUpdated = true;
+        });
+        try {
+            if (scope.robotADirection !== null &&
+                this._model.robotADirection !== scope.robotADirection)
+                this._model.set_property('robotADirection', scope.robotADirection);
+            if (scope.robotBDirection !== null &&
+                this._model.robotBDirection !== scope.robotBDirection)
+                this._model.set_property('robotBDirection', scope.robotBDirection);
+        } finally {
+            this._model.disconnect(tempHandler);
+            GObject.signal_handler_unblock(this._model, this._notifyHandler);
+        }
+
+        if (guiUpdated)
+            SoundServer.getDefault().play('hack-toolbox/update-gui');
+    }
+
+    _errorRecordAtAssignmentLocation(variable, message, value) {
+        const {start, end} = this._variablesCodeview.findAssignmentLocation(variable);
+        return {start, end, message, fixme: String(value)};
+    }
+
+    _searchForErrors(scope) {
+        const errors = [];
+        const PROPS_DEFAULTS = {
+            robotADirection: 'up',
+            robotBDirection: 'up',
+        };
+        Object.entries(PROPS_DEFAULTS).forEach(([prop, defaultValue]) => {
+            if (typeof scope[prop] !== 'string') {
```

Comment:
I'd suggest changing this to check whether the value is `'up'` or `'down'` and change the error message to say that the value must be up or down, since we don't allow just any string. (And we don't care if they use the string "up" or the variable "up")